wally-registry-backend/src/storage/mod.rs

```diff
@@ -32,6 +32,7 @@ pub enum StorageMode {
     #[cfg(feature = "s3-storage")]
     S3 {
         bucket: String,
+        cache_size: Option<u64>,
     },
```

Comment:
I think it's good to follow the serde kebab-case config style. This means in your storage config you will need to change `cache_size` to `cache-size`. 
```suggestion
    #[cfg(feature = "s3-storage")]
    #[serde(rename_all = "kebab-case")]
    S3 {
        bucket: String,
        cache_size: Option<u64>,
    },
```